packages/plugin-ext/src/main/browser/debug/plugin-debug-service.ts

```diff
@@ -104,52 +100,71 @@ export class PluginDebugService implements DebugService, PluginDebugAdapterContr
     }
 
     async provideDebugConfigurations(debugType: keyof DebugRequestTypes, workspaceFolderUri: string | undefined): Promise<theia.DebugConfiguration[]> {
-        const contributor = this.contributors.get(debugType);
-        if (contributor) {
-            return contributor.provideDebugConfigurations && contributor.provideDebugConfigurations(workspaceFolderUri) || [];
-        } else {
+        const pluginProviders =
+            Array.from(this.configurationProviders.values()).
+                filter(
+                    p => p.triggerKind === DebugConfigurationProviderTriggerKind.Initial &&
+                    ( p.type === debugType || p.type === '*') &&
+                    !!p.provideDebugConfigurations);
```

Comment:
The formatting looks weird here:

```suggestion
            Array.from(this.configurationProviders.values()).filter(p => (
                p.triggerKind === DebugConfigurationProviderTriggerKind.Initial &&
                (p.type === debugType || p.type === '*') &&
                p.provideDebugConfigurations
            ));
```